Sources/SwiftlyCore/FileLock.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import SystemPackage
+
+enum FileLockError: Error {
+    case cannotAcquireLock
+    case timeoutExceeded
+}
+
+/// A non-blocking file lock implementation using file creation as locking mechanism.
+/// Use case: When installing multiple Swiftly instances on the same machine,
+/// one should acquire the lock while others poll until it becomes available.
+public struct FileLock {
+    let filePath: FilePath
+
+    public static let defaultPollingInterval: TimeInterval = 1
+    public static let defaultTimeout: TimeInterval = 300.0
+
+    public init(at path: FilePath) throws {
+        self.filePath = path
+        do {
+            let fileURL = URL(fileURLWithPath: self.filePath.string)
+            let contents = getpid().description.data(using: .utf8) ?? Data()
```

Comment:
Can you use https://developer.apple.com/documentation/foundation/processinfo/processidentifier instead of `getpid()`? One fewer obstacle towards getting swiftly to work on Windows.

Reply:
Refactored @jakepetroules